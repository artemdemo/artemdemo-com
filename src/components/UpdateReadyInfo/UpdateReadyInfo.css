.update-ready-info {
  position: fixed;
  cursor: pointer;
  color: #002c50;
  padding: 5px 8px;
  border-radius: 3px;
  right: 10%;
  font-size: 85%;
  box-shadow: rgba(0, 0, 0, 0.25) 1px 1px 4px;
  border: 1px solid rgb(112, 175, 224);
  background-color: rgb(145, 206, 255);
  transition: right 1s;
}
