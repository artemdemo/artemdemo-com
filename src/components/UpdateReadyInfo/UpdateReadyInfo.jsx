import React from 'react';
import { createNanoEvents } from 'nanoevents';
import styled from 'styled-components';

const updateReadyEmitter = createNanoEvents();
const UPDATE_READY_INFO = 'UPDATE_READY_INFO';

export const showUpdateReadyInfo = () => {
    updateReadyEmitter.emit(UPDATE_READY_INFO);
};

const Info = styled.div`
    position: fixed;
    cursor: pointer;
    color: #002c50;
    padding: 5px 8px;
    border-radius: 3px;
    right: 10%;
    font-size: 85%;
    box-shadow: rgba(0, 0, 0, 0.25) 1px 1px 4px;
    border: 1px solid rgb(112, 175, 224);
    background-color: rgb(145, 206, 255);
    transition: right 1s;
`;

class UpdateReadyInfo extends React.PureComponent {
    constructor(props) {
        super(props);

        this.state = {
            updateIsReady: false,
        };

        this.unbind = null;
    }

    componentDidMount() {
        this.unbind = updateReadyEmitter.on(UPDATE_READY_INFO, () => {
            this.setState({
                updateIsReady: true,
            });
        })
    }

    componentWillUnmount() {
        this.unbind();
    }

    handleAppUpdate() {
        window.location.reload();
    }

    render() {
        return (
            <Info
                style={{
                    right: this.state.updateIsReady ? '10%' : '-100%',
                }}
                onClick={this.handleAppUpdate}
            >
                <div>This site has been updated.</div>
                <div>Reload to display the latest version?</div>
                <div>(click here)</div>
            </Info>
        );
    }
}

export default UpdateReadyInfo;
