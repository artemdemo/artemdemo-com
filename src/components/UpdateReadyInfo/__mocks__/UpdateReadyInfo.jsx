import React from 'react';

const UpdateReadyInfo = props => (
    <div data-mock='UpdateReadyInfo'>
        {JSON.stringify(props, null, 2)}
    </div>
);

export default UpdateReadyInfo;
