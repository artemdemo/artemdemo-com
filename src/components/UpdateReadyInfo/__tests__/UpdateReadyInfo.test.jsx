import React from 'react';
import renderer from 'react-test-renderer';
import UpdateReadyInfo from '../UpdateReadyInfo';

describe('UpdateReadyInfo', () => {
    it('should render default', () => {
        const tree = renderer.create(
            <UpdateReadyInfo />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render update is ready', () => {
        const tree = renderer.create(
            <UpdateReadyInfo />
        );

        tree.root.instance.setState({
            updateIsReady: true,
        });

        expect(tree.toJSON()).toMatchSnapshot();
    });
});
