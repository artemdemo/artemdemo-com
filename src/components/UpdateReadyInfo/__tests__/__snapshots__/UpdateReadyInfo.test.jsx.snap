// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateReadyInfo should render default 1`] = `
<div
  className="UpdateReadyInfo__Info-sc-1b2xarw-0 coIosJ"
  onClick={[Function]}
  style={
    Object {
      "right": "-100%",
    }
  }
>
  <div>
    This site has been updated.
  </div>
  <div>
    Reload to display the latest version?
  </div>
  <div>
    (click here)
  </div>
</div>
`;

exports[`UpdateReadyInfo should render update is ready 1`] = `
<div
  className="UpdateReadyInfo__Info-sc-1b2xarw-0 coIosJ"
  onClick={[Function]}
  style={
    Object {
      "right": "10%",
    }
  }
>
  <div>
    This site has been updated.
  </div>
  <div>
    Reload to display the latest version?
  </div>
  <div>
    (click here)
  </div>
</div>
`;
